pkg/route/validation/validation.go

```diff
@@ -342,6 +370,41 @@ func validateWildcardPolicy(host string, policy routev1.WildcardPolicyType, fldP
 	return nil
 }
 
+var (
+	notAllowedHTTPHeaderResponse    = []string{"strict-transport-security"}
+	notAllowedHTTPHeaderResponseSet = sets.NewString(notAllowedHTTPHeaderResponse...)
+)
+
+func validateHeaders(path *field.Path, headerValues []routev1.RouteHTTPHeader) *field.Error {
+	for key, value := range headerValues {
+		if value.Action.Set != nil && len(value.Action.Set.Value) != 0 && len(value.Action.Set.Value) > 4096 {
+			return field.Invalid(path.Index(key).Child("action.set.value"), value.Name, fmt.Sprintf("value exceeds the maximum length which is 2048"))
+		}
+		if value.Action.Set != nil && len(value.Action.Set.Value) == 0 {
+			return field.Invalid(path.Index(key).Child("action.set.value"), value.Name, fmt.Sprintf("minimum length of value shall be one."))
+		}
+		if path.String() == "spec.httpHeaders.actions.response" && value.Action.Set != nil && len(value.Action.Set.Value) != 0 {
+			r := regexp.MustCompile("^(?:%(?:%|(?:\\{[-+]?[QXE](,[-+]?[QXE])*\\})?\\[(?:res\\.hdr\\([0-9A-Za-z-]+\\)|ssl_c_der)(?:,(?:lower|base64))*\\])|[^%[:cntrl:]])*$")
```

Comment:
I'd like to see the full list of details that map to this regex pattern.  It is not in the enhancement proposal or the API godoc.  It's hard to review the unit testing without a full sense of what needs to be tested.

Reply:
The EP has been updated. 
The valid test cases for dynamic request header values Eg- "%[req.hdr(host),lower]"   "%{+Q}[ssl_c_der,base64]"